File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        
        prefix = postfix = 1
        for i in range(len(nums)):
            res.append(prefix)
            prefix *= nums[i]

        for i in range(len(nums) - 1, -1, -1):
            res[i] = res[i] * postfix
            postfix *= nums[i]

        return res
        

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> prefixSum = new HashMap<>();
        prefixSum.put(0, 1);
        int curSum = 0, res = 0;

        for (int num : nums) {
            curSum += num;
            int diff = curSum - k;

            res += prefixSum.getOrDefault(diff, 0);
            prefixSum.put(curSum, prefixSum.getOrDefault(curSum, 0) + 1);
        }
        return res;
    }
}